indexing-service/src/test/java/org/apache/druid/indexing/overlord/TaskLockboxTest.java

```diff
@@ -1272,6 +1276,627 @@ public void testGetLockedIntervalsForEqualPriorityTask() throws Exception
     );
   }
 
+  @Test
+  public void testExclusiveLockCompatibility() throws Exception
+  {
+    final List<Task> tasks = new ArrayList<>();
+
+    final TaskLock theLock = insertTaskLock(
+        TaskLockType.EXCLUSIVE,
+        Intervals.of("2017/2018"),
+        MEDIUM_PRIORITY,
+        tasks
+    );
+    Assert.assertNotNull(theLock);
+    Assert.assertFalse(theLock.isRevoked());
+
+    // Another exclusive lock cannot be created for an overlapping interval
+    Assert.assertNull(
+        insertTaskLock(
+            TaskLockType.EXCLUSIVE,
+            Intervals.of("2017-05-01/2017-06-01"),
+            MEDIUM_PRIORITY,
+            tasks
+        )
+    );
+
+    // A shared lock cannot be created for an overlapping interval
+    Assert.assertNull(
+        insertTaskLock(
+            TaskLockType.SHARED,
+            Intervals.of("2016/2019"),
+            MEDIUM_PRIORITY,
+            tasks
+        )
+    );
+
+    // A replace lock cannot be created for an overlapping interval
+    Assert.assertNull(
+        insertTaskLock(
+            TaskLockType.REPLACE,
+            Intervals.of("2017/2018"),
+            MEDIUM_PRIORITY,
+            tasks
+        )
+    );
+
+    // An append lock cannot be created for an overlapping interval
+    Assert.assertNull(
+        insertTaskLock(
+            TaskLockType.APPEND,
+            Intervals.of("2017-05-01/2018-05-01"),
+            MEDIUM_PRIORITY,
+            tasks
+        )
+    );
+
+    final Set<TaskLock> expectedLocks = ImmutableSet.of(
+        theLock
+    );
+    Assert.assertEquals(expectedLocks, getAllLocks(tasks));
+  }
+
+  @Test
+  public void testExclusiveLockCanRevokeAllIncompatible() throws Exception
+  {
+    final List<Task> tasks = new ArrayList<>();
+
+    // Revoked shared lock of higher priority -> revoked locks are not incompatible
+    final TaskLock sharedLock = insertTaskLock(
+        TaskLockType.SHARED,
+        Intervals.of("2016/2019"),
+        HIGH_PRIORITY,
+        tasks
+    );
+    Assert.assertNotNull(sharedLock);
+    lockbox.revokeLock(tasks.get(0).getId(), sharedLock);
+
+    // Active Exclusive lock of lower priority -> will be revoked
+    final TaskLock exclusiveLock = insertTaskLock(
+        TaskLockType.EXCLUSIVE,
+        Intervals.of("2017-01-01/2017-02-01"),
+        LOW_PRIORITY,
+        tasks
+    );
+    Assert.assertNotNull(exclusiveLock);
+    Assert.assertFalse(exclusiveLock.isRevoked());
```

Comment:
Why will this be revoked?  the shared lock that it would conflict with was revoked, in which case this should be fine, no?

Reply:
Oooooh, I see, it's asserting that the lock was granted and the revoked is false...  I totally read it wrong, partially because your comment made me think that you were testing that it was going to be revoked.